src/common/rtc.cc

```diff
@@ -28,6 +32,43 @@
 namespace mxnet {
 namespace rtc {
 
+namespace {
+
+  const char* gpu_arch_prefix = "--gpu-architecture=compute_";
+  const std::locale locale;
+
+  bool FindGpuArchOption(std::string option) {
+    auto lower = [&](char c) { return std::tolower(c, locale); };
+    std::transform(option.begin(), option.end(), option.begin(), lower);
+    return option.find(gpu_arch_prefix) != std::string::npos;
+  }
+
+  void SetGpuArch(std::vector<const char *>* c_options, std::string* arch_string) {
+    assert(c_options != nullptr);
+    assert(arch_string != nullptr);
+
+    // Check options for explicit device architecture information.
+    for (const auto& option : *c_options) {
+      if (FindGpuArchOption(option)) {
+        return;
+      }
+    }
+
+    // If not present, use the first gpu device present for architecture information.
+    int num_devices;
+    CUDA_CALL(cudaGetDeviceCount(&num_devices));
+
+    if (num_devices > 0) {
+      cudaDeviceProp prop;
+      CUDA_CALL(cudaGetDeviceProperties(&prop, 0));
```

Comment:
what if there are multiple gpus with different arch?

Reply:
how about we just use CUDA_ARCH from Makefile.

Reply:
Thanks for the review Eric.  I see your point about supported multiple device types, however I'm not sure if CUDA_ARCH's are the best place to configure this.  My reasoning is firstly that this is part of the Makefile, not a user visible configuration option, so users may not intuitively know where to look to set this.  Secondly I think this is something should should be specified at runtime, not MXNet build time.

I would propose that we change the constructor of [Chunk](https://github.com/KellenSunderland/incubator-mxnet/blob/0b11c7eadcacf124b8205f4215b38d5cc6244bf0/src/common/rtc.cc#L93)  such that it compiles a ptx for each gen of device on the system.  We then change the [ptx_*](https://github.com/apache/incubator-mxnet/blob/master/include/mxnet/rtc.h#L63) member to a mapping of device id's to ptx*s.  We're already passing ctx and referring to device ids when we load a module [here](https://github.com/KellenSunderland/incubator-mxnet/blob/0b11c7eadcacf124b8205f4215b38d5cc6244bf0/src/common/rtc.cc#L128), so it's natural that we should specify the correct ptx at that time.

Reply:
During build only archs in CUDA_ARCH are supported. I makes sense for RTC kernels to support the same set of devices as the builtin kernels.

Reply:
We could use these values, but I'm still not convinced of why this would be necessary.  This is specifically a runtime feature right?  At runtime we can detect which GPUs are present and compile specifically for those.  Would it bring users any value to compile only for pre-configured values?  

Let's assume a user has a device that has sm_72, and they've pip installed mxnet 12.1, which for the sake of argument has a max of sm_70 in it's CUDA_ARCH.  Why would we want to prevent this user from using rtc with 72?

Reply:
Because all the other kernels wouldn't work anyway. Doesn't matter whether rtc works or not

Reply:
If you want to enumerate the present GPUs thats also fine.
But no need to make multiple ptx. A single ptx can support multiple archs